python/entry_point/generator_test.py

```diff
@@ -0,0 +1,190 @@
+                run(
+                    entry_points=entry_points,
+                    out=pathlib.Path(__file__),
+                    console_script=None,
+                    shebang="#!/dev/null",
+                )
+
+        self.assertEqual(
+            "Please select one of the following console scripts: foo, bar",
+            cm.exception.args[0],
+        )
+
+    def test_a_single_entry_point(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir = pathlib.Path(tmpdir)
+            given_contents = (
+                textwrap.dedent(
+                    """
+            [console_scripts]
+            foo = foo.bar:baz
+            """
+                ).strip()
+                + "\n"
+            )
+            entry_points = tmpdir / "entry_points.txt"
+            entry_points.write_text(given_contents)
+            out = tmpdir / "out.py"
+
+            run(
+                entry_points=entry_points,
+                out=out,
+                console_script=None,
+                shebang="#!/dev/null",
+            )
+
+            got = out.read_text()
+
+        want = textwrap.dedent(
+            """\
+        #!/dev/null
+        import sys
+        # Drop the first entry in the sys.path, because it will point to our workspace
+        # where we are generating the entry_point script and it seems that some package
+        # break due to this reason (e.g. pylint). This means that we should not use this
+        # script to ever generate entry points for scripts within the main workspace,
+        # but that is fine, we can create a separate generator or a boolean flag for
+        # that.
+        if ".runfiles" not in sys.path[0]:
+            sys.path = sys.path[1:]
+
+        try:
+            from foo.bar import baz
+        except ImportError:
+            entries = "\\n".join(sys.path)
+            print("Printing sys.path entries for easier debugging:")
+            print(f"sys.path is:\\n{entries}")
+
+        if __name__ == "__main__":
+            sys.exit(baz())
+        """
+        )
+        self.assertEqual(want, got)
+
+    def test_a_second_entry_point_class_method(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir = pathlib.Path(tmpdir)
+            given_contents = (
+                textwrap.dedent(
+                    """
+            [console_scripts]
+            foo = foo.bar:Bar.baz
+            bar = foo.baz:Bar.baz
+            """
+                ).strip()
+                + "\n"
+            )
+            entry_points = tmpdir / "entry_points.txt"
+            entry_points.write_text(given_contents)
+            out = tmpdir / "out.py"
+
+            run(
+                entry_points=entry_points,
+                out=out,
+                console_script="bar",
+                shebang="#!/dev/null",
+            )
+
+            got = out.read_text()
+
+        want = textwrap.dedent(
```

Comment:
Asserting the exact content for one test case is fine; it lets one see what the generated code is supposed to look like a glance.

For variations of the base case, though, just search for the substrings of interest.

Otherwise, if the template is changed, we have to update more of the tests unnecessarily.